File: core/src/main/java/com/ustadmobile/core/impl/UmCallbackUtil.java
package com.ustadmobile.core.impl;

public class UmCallbackUtil {

    public static <T> void onSuccessIfNotNull(UmCallback<T> callback, T result) {
        if(callback != null)
            callback.onSuccess(result);
    }

    public static <T> void onFailIfNotNull(UmCallback<T> callback, Throwable exception) {
        if(callback != null)
            callback.onFailure(exception);
    }
}

File: lib-util/src/main/java/com/ustadmobile/core/impl/UmCallbackUtil.kt
package com.ustadmobile.core.impl

object UmCallbackUtil {

    fun <T> onSuccessIfNotNull(callback: UmCallback<T>?, result: T) {
        callback?.onSuccess(result)
    }

    fun <T> onFailIfNotNull(callback: UmCallback<T>?, exception: Throwable) {
        callback?.onFailure(exception)
    }
}
